backend/site-api/src/db/custom_data_repo.rs

```diff
@@ -27,6 +35,68 @@ pub struct CustomDataRepo {
     pub manifest_dir: String,
 }
 
+fn row_to_list_result(row: SqliteRow) -> Result<(BTreeMap<String, String>, i64), Error> {
+    let count = row.try_get("count")?;
+    let entity = map_to_key_value(row)?;
+    Ok((entity, count))
+}
+
+fn map_to_key_value(row: SqliteRow) -> Result<BTreeMap<String, String>, Error> {
+    let mut row_data = BTreeMap::new();
+
+    for column in row.columns().iter() {
+        let column_name = column.name();
+        match column_name {
+            "count" => continue, // Skip processing for "count" column
+            "id" => {
+                let value: String = row.try_get::<i64, _>(column_name)?.to_string();
+                row_data.insert(column_name.to_string(), value);
+            }
+            _ => {
+                let value: String = row.try_get(column_name)?; // Assuming all values are strings
+                row_data.insert(column_name.to_string(), value);
+            }
+        }
+    }
+
+    Ok(row_data)
+}
+
+fn map_custom_data_sqlx_err(e: sqlx::Error) -> DbError {
+    let err_str = e.to_string();
+    match e {
+        sqlx::Error::Database(err) => {
+            let constraint_info = err.message().split(": ").collect::<Vec<&str>>();
+
+            if err.code() == Some(Borrowed("2067")) {
```

Comment:
The reason I checked with `err.code()` is because `err.constraint()` returns `None` in this case, so I use `code()` instead.